addons/sale/data/mail_template_data.xml

```diff
@@ -4,220 +4,223 @@
+            <field name="partner_to">{{ object.partner_id.id }}</field>
             <field name="body_html" type="html">
 <div style="margin: 0px; padding: 0px;">
     <p style="margin: 0px; padding: 0px; font-size: 12px;">
         Hello,
         <br/><br/>
-        % set transaction = object.get_portal_last_transaction()
-        Your order <strong>${object.name}</strong> amounting in <strong>${format_amount(object.amount_total, object.currency_id)}</strong>
-        % if object.state == 'sale' or (transaction and transaction.state in ('done', 'authorized')) :
+        <t t-set="transaction" t-value="object.get_portal_last_transaction()"/>
+        Your order <strong t-out="object.name or ''">S00049</strong> amounting in <strong t-out="format_amount(object.amount_total, object.currency_id) or ''">$ 10.00</strong>
+        <t t-if="object.state == 'sale' or (transaction and transaction.state in ('done', 'authorized'))">
             has been confirmed.<br/>
             Thank you for your trust!
-        % elif transaction and transaction.state == 'pending' :
+        </t>
+        <t t-elif="transaction and transaction.state == 'pending'">
             is pending. It will be confirmed when the payment is received.
-            % if object.reference:
-                Your payment reference is <strong>${object.reference}</strong>.
-            % endif
-        % endif
+            <t t-if="object.reference">
+                Your payment reference is <strong t-out="object.reference or ''"></strong>.
+            </t>
+        </t>
         <br/><br/>
         Do not hesitate to contact us if you have any questions.
         <br/><br/>
     </p>
-% if object.website_id:
+<t t-if="hasattr(object, 'website_id') and object.website_id">
     <div style="margin: 0px; padding: 0px;">
         <table width="100%" style="color: #454748; font-size: 12px; border-collapse: collapse;">
             <tr style="border-bottom: 2px solid #dee2e6;">
                 <td style="width: 150px;"><strong>Products</strong></td>
                 <td></td>
                 <td width="15%" align="center"><strong>Quantity</strong></td>
                 <td width="20%" align="right"><strong>
-                % if object.user_id.has_group('account.group_show_line_subtotals_tax_excluded'):
+                <t t-if="object.user_id.has_group('account.group_show_line_subtotals_tax_excluded')">
                     VAT Excl.
-                % else
+                </t>
+                <t t-else="">
                     VAT Incl.
-                % endif
+                </t>
                 </strong></td>
             </tr>
         </table>
-        % for line in object.order_line:
-            % if not line.is_delivery and line.display_type in ['line_section', 'line_note']:
+        <t t-foreach="object.order_line" t-as="line">
+            <t t-if="(not hasattr(line, 'is_delivery') or not line.is_delivery) and line.display_type in ['line_section', 'line_note']">
```

Comment:
```suggestion
            <t t-if="not line._is_delivery() and line.display_type in ['line_section', 'line_note']">
```

Reply:
> Little comment for the mail templates in sale.
> 
> Why is there all those "demo" values in the templates ? thinking

Because it's the easiest way technically to have demo data on the templates.
This is a controversial subject.

> (I love the feature, but it's quite late for such a breaking change post-freeze boom)

There is a big value having the templates in v15 so it's a price to pay.

> (Next step: support mail templates extension with xpath ?)

If there is a need, it would not be difficult.